### PR TITLE
rllab ec2 tutorial

### PR DESCRIPTION
**Pull request by nskh**
draft tutorial is in this commit. let me know what you think.

also - an idea: these shouldn't be tutorials, but should go into our documentation. they can be easily converted to .rst files and would look nice in our docs. thoughts?

(a copy of https://github.com/berkeleyflow/flow/pull/70)
(how would i remove the number of commits here?)

### CODE REVIEW

**Comment by AboudyKreidieh** [comment]
don't worry, commits are fine (they reflect our actual commit history). Two comments though:
1. I think some stuff are left behind that should be there (e.g. the warnings folder). Can you please delete them?
2. You mentioned turning this into a document in readthedocs instead. Do you wanna do that for this PR?

**Comment by nskh** [comment]
@AboudyKreidieh - deleted the old files and added a page to the documentation. Not 100% where to put this tutorial... we might need some sort of subfolder structure for our docs, in which case we could put Setup Instructions, Tutorials, and the rllib+rllab tutorials inside those. @eugenevinitsky, thoughts?

**Comment by eugenevinitsky** [comment]
Hey nish, can you add the MakeFile to this?

**Comment by eugenevinitsky** [comment]
Want to review it but I can't make it.

**Comment by nskh** [comment]
@eugenevinitsky the Makefile should already be there - are you running `make html` from inside `docs/`?

**Comment by eugenevinitsky** [comment]
Huh I guess it is, looking at this PR again

File: docs/source/rllabec2.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+EC2 instance. This tutorial assumes rllab has been installed correctly
+(`instructions <https://rllab.readthedocs.io/en/latest/user/installation.html>`_).
+
+Setting up rllab with AWS
+-------------------------
+
+First, follow the `rllab cluster setup
+instructions <https://rllab.readthedocs.io/en/latest/user/cluster.html>`__
+with region ``us-west-1``. Modify ``rllab/config_personal.py`` to
+reference the most current Flow Docker image.
```
Comment (eugenevinitsky) [inline docs/source/rllabec2.rst]: can we add the name of the most current Flow docker image? @kjang96 which one are you using?

File: docs/source/rllabec2.rst [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+Navigate to your ``flow`` directory and modify ``Makefile.template`` per
+the instructions in that file. The variable ``RLLABDIR`` should be the
+relative path from your ``flow`` directory to ``rllab`` and should not
+have a backslash at the end.
```
Comment (eugenevinitsky) [inline docs/source/rllabec2.rst]: Bold "should not have a backslash at the end"

File: docs/source/rllabec2.rst [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+available to the EC2 instance. Thus, commands are included to copy over
+your ``flow`` directory to your ``rllab`` root directory (this is the
+reason for the ``RLLABDIR`` variable above).
+
+-  Before running an experiment, run ``make prepare`` from your ``flow``
+   directory.
+-  Ensure you have committed or otherwise tracked the state of your
+   ``flow`` directory, because that instance is what will be used to run
+   your experiment. Upon visualization, the same files will need to be
+   used—for example, changes to environment's state-space would break
```
Comment (eugenevinitsky) [inline docs/source/rllabec2.rst]: "changes to your environment's state-space"

File: docs/source/rllabec2.rst [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+   the ability to run a trained policy using a different state space.
+   Check out an old commit of your ``flow`` directory and run
```
Comment (eugenevinitsky) [inline docs/source/rllabec2.rst]: I don't believe you need to run make prepare to use the visualization tools?"

File: docs/source/rllabec2.rst [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+``make clean`` removes the debug directory and also all XML files in
+rllab root directory to reduce the size of the data to upload to AWS. If
+you are using already-existing network files (from, say, OpenStreetMap),
+ensure they do not get deleted by ``make clean`` by storing such files
+elsewhere.
+
+Inside the experiment, change the ``mode`` to ``ec2`` (other options are
+``local`` and ``local_docker``). You should run the experiment in
+``local_docker`` mode briefly before running the ``ec2`` version to
```
Comment (eugenevinitsky) [inline docs/source/rllabec2.rst]: Can you add a descripton of what running in local_docker tests?

**Comment by eugenevinitsky** [comment]
@nskh I like the idea of a tutorials substructure in the docs, can you create one and add this to it? I'm making an rllib tutorial and will include it there as well.

File: docs/source/rllabec2.rst (hunk 1 above)
Comment (nskh) [reply docs/source/rllabec2.rst]: @fywu85 @kjang96 @lucfisc any updates on the docker image to reference?

File: docs/source/rllabec2.rst (hunk 5 above)
Comment (nskh) [reply docs/source/rllabec2.rst]: What do you think I should say here?

**Comment by nskh** [comment]
@eugenevinitsky updated this but have some questions before merging - let me know when you see this.

File: docs/source/rllabec2.rst (hunk 1 above)
Comment (nskh) [reply docs/source/rllabec2.rst]: fywu85/flow

Comment (nskh) [reply docs/source/rllabec2.rst]: @fywu85 @lucfisc @kjang96 we should figure out what docker image to mention here

**Pull request merged**